src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinder.cs

```diff
@@ -448,17 +448,12 @@ protected virtual void CopyToModel(object target, IEnumerable<TElement> sourceCo
 
         private static IEnumerable<string> GetIndexNamesFromValueProviderResult(ValueProviderResult valueProviderResult)
         {
-            IEnumerable<string> indexNames = null;
-            if (valueProviderResult != null)
+            var indexes = (string[])valueProviderResult;
+            if (indexes != null && indexes.Length > 0)
```

Comment:
Isn't it impossible for `indexes` to be null here? `valueProviderResult` is a struct and the `(string[])` must cast either succeed or throw.

Reply:
Might be `null` for `default(StringValues)`

Reply:
Yowzers! Return null from an implicit conversion sounds pretty evil. We couldn't even return a static empty array?

Reply:
No, that's expected to preserve transitivity. E.g. (string[])(StringValues)null;